File: yourOrderPlease/yourOrderPlease.js
function order(words){
  //split by ' ', make push words into array, split each word and identify number, fetch index and order both arrays by ascending order
  
  const tempMemory = words.split(' ');
  console.log(tempMemory);
  let sorted = [];
  let result = [];
  
  tempMemory.forEach(function(element, i){
    let tempArr = element.split('');
    let elementMem = [];
    let index;
    tempArr.forEach(function(character, ii){
      elementMem.push(character);
      if (!isNaN(Number(character))){
        index = character
        console.log(index);
      }
    })
   let resElement = elementMem.join('');
   result[index - 1] = resElement;
  })
  return result.join(' ');
}

order("4of Fo1r pe6ople g3ood th5e the2")
order("is2 Thi1s T4est 3a")


//Best Solution:
function order(words){
  
  return words.split(' ').sort(function(a, b){
      return a.match(/\d/) - b.match(/\d/);
   }).join(' ');
}   

File: yourOrderPlease/yourOrderPlease.py
def order(sentence):
  words_in_list = sentence.split()
  ordered_words_in_increasing_number = sorted(words_in_list, key = numbers_from_words)
  return (" ").join(ordered_words_in_increasing_number) 

def numbers_from_words(word):
  for letter in word:
    if letter.isdigit():
      return int(letter) #will return numbers from here and this number with decide the order in sorted function, for eg if we return 1 from here that will come first in sorted function
  return None